File: docs/competitors/lab1/php/SumNumbers.php
<?php
declare(strict_types=1);

define('NUMBER_DELIMITER', ' ');
define('INPUT_MESSAGE', "Введите числа через пробел: ");
define('RESULT_MESSAGE', "Сумма введенных чисел: ");
define('NEW_LINE', "\n");

//что делать, если пользователь ввел не число? 
//ошибка и прекращение работы или просто пропуск?
//я сделала просто пропуск
function main() 
{
    echo INPUT_MESSAGE;
    $numbers = readNumbers();
    $sum = 0;

    foreach ($numbers as $number) 
    {
        if (is_numeric($number)) 
        {
            $sum += floatval($number);
        }
    }
    echo RESULT_MESSAGE . $sum . NEW_LINE;
}

/**
 * @return string[]
 */
function readNumbers(): array 
{
    $input = trim(fgets(STDIN));
    return explode(NUMBER_DELIMITER, $input);
}

main();
?>

File: docs/competitors/lab1/go/SumNumbers/SumNumbers.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

const (
	inputMessage    = "Введите числа через пробел: "
	resultMessage   = "Сумма введенных чисел: "
	newLineSymbol   = '\n'
	numberDelimiter = " "
)

func main() {
	fmt.Print(inputMessage)
	numbers := readNumbers()
	sum := 0.0

	for _, numberStr := range numbers {
		if number, err := strconv.ParseFloat(numberStr, 64); err == nil {
			sum += number
		}
	}
	
	fmt.Printf("%s%.2f\n", resultMessage, sum)
}

func readNumbers() []string {
	reader := bufio.NewReader(os.Stdin)

	input, _ := reader.ReadString(newLineSymbol)
	input = strings.TrimSpace(input)
	
	return strings.Split(input, numberDelimiter)
}